src/hash.cr

```diff
@@ -1747,6 +1747,28 @@ class Hash(K, V)
     true
   end
 
+  # Returns `true` if `self` is a subset of *other*.
+  def proper_subset_of?(other : Hash)
+    return false if other.size <= size
+    all? { |key, value| other[key]? == value }
```

Comment:
With `[]?` there's no differentiation between `nil` value and no value. Thus `{"a" => nil}.subset_of?({"b" => nil}) # => true`.

`other.fetch(key) { return false }` should fix this.

Reply:
It probably wouldn't hurt to add specs for such edge cases.

Reply:
Done